idunn/datasources/pages_jaunes.py

```diff
@@ -57,6 +60,33 @@ def format_where(bbox):
         left, bot, right, top = bbox
         return f"gZ{left:.6f},{bot:.6f},{right:.6f},{top:.6f}"
 
+    @classmethod
+    def fetch_search(cls, query: SearchQueryParams, intentions=None, is_france_query=False):
+        if len(intentions) > 0:
+            return asyncio.create_task(
+                run_in_threadpool(
+                    pj_source.search_places,
+                    query,
+                    intentions[0].description._place_in_query,
+                ),
+                name="ia_fetch_pj",
+            )
+        return asyncio.create_task(
+            run_in_threadpool(
```

Comment:
In a similar fashion, I would have said that it should be left to the caller but maybe we should keep this and make the method async in the `Datasource` parent class.